load/app/subsidies_app.go

```diff
@@ -0,0 +1,172 @@
+	if err != nil {
+		return nil, err
+	}
+
+	subsidiesFund := new(big.Int).Mul(big.NewInt(10_000), big.NewInt(1e18))
+	err = appContext.FundAccounts([]common.Address{sponsorAccount.address}, subsidiesFund)
+	if err != nil {
+		return nil, fmt.Errorf("failed to sponsor account; %w", err)
+	}
+
+	// Deploy the Counter counterContract to be used by this application.
+	counterContract, receipt, err := DeployContract(appContext, contract.DeployCounter)
+	if err != nil || receipt.Status != types.ReceiptStatusSuccessful {
+		return nil, errors.Join(fmt.Errorf("failed to deploy Counter contract"), err)
+	}
+	counterContractAddress := receipt.ContractAddress
+
+	// Mount registry contract object from the system address
+	subsidiesRegistry, err := registry.NewRegistry(registry.GetAddress(), client)
+	if err != nil {
+		return nil, fmt.Errorf("failed to bind to subsidies registry contract; %w", err)
+	}
+
+	// Fund a subsidy for the counter contract:
+	// 1. Get the fund ID for the counter contract
+	_, fundId, err := subsidiesRegistry.ContractSponsorshipFundId(nil, counterContractAddress)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read fundId from contract; %w", err)
+	}
+
+	// 2. Get the transact opts for the sponsor account
+	opts, err := appContext.GetTransactOptions(sponsorAccount)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get transact opts for sponsorship; %w", err)
+	}
+	// 3. Fund with all available balance minus gas for the sponsorship transaction
+	opts.Value = new(big.Int).Sub(subsidiesFund,
+		new(big.Int).Mul(opts.GasPrice, big.NewInt(150_000)))
+
+	// 4. Call the sponsorship method and check receipt for sanity
+	tx, err := subsidiesRegistry.Sponsor(opts, fundId)
+	if err != nil {
+		return nil, fmt.Errorf("failed to sponsor contract; %w", err)
+	}
+	receipt, err = appContext.GetReceipt(tx.Hash())
+	if err != nil || receipt.Status != types.ReceiptStatusSuccessful {
+		return nil, errors.Join(fmt.Errorf("sponsorship transaction failed"), err)
+	}
+
+	return &SubsidiesApplication{
+		counterContract: counterContract,
+		accountFactory:  accountFactory,
+	}, nil
+}
+
+func (f *SubsidiesApplication) CreateUsers(appContext AppContext, numUsers int) ([]User, error) {
+	users := make([]User, numUsers)
+	addresses := make([]common.Address, numUsers)
+	for i := range users {
+		// Generate a new account for each worker - avoid account nonces related bottlenecks
+		workerAccount, err := f.accountFactory.CreateAccount(appContext.GetClient())
+		if err != nil {
+			return nil, err
+		}
+		users[i] = &SubsidiesUser{
+			counterContract: f.counterContract,
+			Sender:          workerAccount,
+		}
+		addresses[i] = workerAccount.address
+	}
+
+	// Fund all accounts with 1 FTM so they can pay for the balance transfer
+	// gas costs shall be covered by the subsidy
+	err := appContext.FundAccounts(addresses, big.NewInt(1e18))
+	if err != nil {
+		return nil, fmt.Errorf("failed to fund accounts; %w", err)
+	}
```

Comment:
Is this still needed? It should be possible for a 0-balanced account to run sponsored transactions, right?